.github/workflows/cd-syft-dev.yml

```diff
@@ -150,3 +162,24 @@ jobs:
           add: "."
           push: "origin main"
           cwd: "./infrastructure/"
+
+      - name: Cleanup Azure Container Registry
+        uses: azure/CLI@v1
+        with:
+          # SKIP_LINES = latest version dev & dev-<sha> (2 lines) + keep "n" previous version (n lines) + 1
+          inlineScript: |
+            ACR_REGISTRY_NAME=${{ secrets.ACR_REGISTRY_NAME }}
+            REPO_LIST=$(az acr repository list -n $ACR_REGISTRY_NAME -o tsv)
+
+            KEEP_PREV_VERSIONS=1
+            TAIL_FROM_LINE=$((2 + $KEEP_PREV_VERSIONS + 1))
+
+            for repo in $REPO_LIST
+            do
+                echo "Cleaning up '$repo'"
+                az acr repository show-tags --name $ACR_REGISTRY_NAME --repository $repo --orderby time_desc --output tsv | tail -n +$TAIL_FROM_LINE | xargs -r -I% az acr repository delete --name $ACR_REGISTRY_NAME --image $repo:% --yes
```

Comment:
To summarize, for each repository (`az acr repository list ...`):
* List all tags, sort them by time (latest first) (`az acr repository show-tags ...`) Outputs:
```
dev-[latest-hash]
dev
dev-[previous-hash]
dev-[older-hash]
dev-[oldest-hash]
dev-[oldest-hash]
...
```

* Skip the first 2 + "k" previous tags (`tail -n ...`)
* Delete everything after that (`az acr repository delete ...`)